provider/clusterautoscaler/cluster_autoscaler_resource.go

```diff
@@ -0,0 +1,605 @@
+/*
+Copyright (c) 2023 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clusterautoscaler
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/listvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	sdk "github.com/openshift-online/ocm-sdk-go"
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	"github.com/terraform-redhat/terraform-provider-rhcs/provider/common"
+)
+
+type ClusterAutoscalerResourceType struct {
+}
+
+type ClusterAutoscalerResource struct {
+	collection *cmv1.ClustersClient
+}
+
+func New() resource.Resource {
+	return &ClusterAutoscalerResource{}
+}
+
+var _ resource.Resource = &ClusterAutoscalerResource{}
+var _ resource.ResourceWithImportState = &ClusterAutoscalerResource{}
+var _ resource.ResourceWithConfigure = &ClusterAutoscalerResource{}
+
+func (r *ClusterAutoscalerResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_cluster_autoscaler"
+}
+
+func (r *ClusterAutoscalerResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
```

Comment:
All attributes but "cluster" are optional, not computed and without default value. What happens when user creates it just with the cluster ID? shouldn't we save the default values in the state?

Reply:
If they are not computed, we should not assign values. Once a user changes a value - it will update the backend - and store the value in the state.

Reply:
Yes, I understand that, but I guess there are default values, and now I understand that the API doesn't return the default values.
So I guess its fine